src/main/java/edu/njit/jerse/automation/AsheAutomation.java

```diff
@@ -0,0 +1,141 @@
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import com.github.javaparser.ast.nodeTypes.NodeWithName;
+import edu.njit.jerse.ashe.ASHE;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+// TODO: Add this functionality to the README
+/**
+ * * The {@code AsheAutomation} class provides automation for processing Java files
+ * * within a given directory. It applies {@code ASHE}'s minimization and correction mechanisms to
+ * * Java files in or under a given directory.
+ */
+public class AsheAutomation {
+    private static final Logger LOGGER = LogManager.getLogger(AsheAutomation.class);
+
+    /**
+     * Calls the {@code processJavaFile} method on each Java file in or under
+     * the given directory.
+     *
+     * @param dir      the directory to iterate over
+     * @param rootPath the root path of the project, used for determining the relative
+     *                 path of each Java file within the project structure
+     * @throws IOException if an I/O error occurs while accessing the directory
+     */
+    public static void iterateJavaFiles(File dir, String rootPath) throws IOException {
+        LOGGER.info("Iterating over Java files in directory: {}", dir.getAbsolutePath());
+
+        try (Stream<Path> paths = Files.walk(Paths.get(dir.getAbsolutePath()))) {
+            paths.filter(Files::isRegularFile)
+                    // include only files with .java extension
+                    .filter(path -> path.toString().endsWith(".java"))
+                    .forEach(path -> {
+                        try {
+                            processJavaFile(path.toFile(), rootPath);
+                        } catch (IOException | ExecutionException | InterruptedException | TimeoutException e) {
+                            LOGGER.error("Error processing Java file: {}", path.toFile().getAbsolutePath(), e);
+                            throw new RuntimeException("Error processing Java file: " + path.toFile().getAbsolutePath(), e);
+                        }
+                    });
+        }
+        LOGGER.info("Completed iterating over Java files in directory: {}", dir.getAbsolutePath());
+    }
+
+    /**
+     * Processes an individual Java file using the ASHE framework. This involves parsing
+     * the file to extract class and method information, and then for each public method,
+     * the ASHE 'run' method is invoked to apply minimization and error correction.
+     *
+     * @param javaFile        the Java file to be processed
+     * @param projectRootPath the root path of the project, used for determining the
```

Comment:
I believe that this must be a prefix of the absolute path of `javaFile`.  I suggest both documenting that, and enforcing it via a run-time check.